src/cocotb/share/lib/vpi/VpiImpl.cpp

```diff
@@ -505,61 +506,79 @@ GpiObjHdl *VpiImpl::native_check_create(int32_t index, GpiObjHdl *parent) {
     return new_obj;
 }
 
-GpiObjHdl *VpiImpl::get_root_handle(const char *name) {
-    vpiHandle root;
+GpiObjHdl *VpiImpl::get_scope_handle(const char *name, bool is_package) {
+    vpiHandle scope;
     vpiHandle iterator;
     GpiObjHdl *rv;
-    std::string root_name;
+    std::string scope_name;
+
+    PLI_INT32 iterator_type;
+    std::string full_name;
+    if (is_package) {
+        iterator_type = vpiInstance;
+        full_name = std::string(name) + "::";
```

Comment:
Note that the standard says "should" not "shall", so the "::" is not required. You should likely support both it being include and just the bare name, though I agree it would be best for the simulator to include it for the reason given in the standard.